google-cloud-storage/lib/google/cloud/storage/file/signer_v4.rb

```diff
@@ -156,25 +152,48 @@ def canonical_query query, algorithm, credential, goog_date, expires,
             query["X-Goog-Credential"] = credential
             query["X-Goog-Date"] = goog_date
             query["X-Goog-Expires"] = expires
-            query["X-Goog-SignedHeaders"] = CGI.escape signed_headers_str
-            query = query.sort_by { |k, _| k.to_s.downcase }.to_h
+            query["X-Goog-SignedHeaders"] = signed_headers_str
+            query = query.map { |k, v| [escape_query_param(k), escape_query_param(v)] }.to_h
```

Comment:
redundant `to_h` since the following line is going to treat it as an Enumerable anyway.